Improved CAF platform selection

APokorny opened the pull request:
Use the build property to decide whether the CAF platform shall be
used. On android 5 the regular android platform is used on CAF devices.

It checks the android build property - if it is >= 7.1.0 it will use CAF, and prohibit it otherwise.
This should resolve issues https://github.com/ubports/ubuntu-touch/issues/494

APokorny committed:
Improved CAF platform selection

Use the build property to decide whether the CAF platform shall be
used. On android 5 the regular android platform is used on CAF devices.

APokorny committed:
tuple constructor selection

mariogrip commented on src/platforms/android/server/platform.cpp:
```diff
@@ -221,12 +223,14 @@ mg::PlatformPriority probe_graphics_platform(mo::ProgramOption const& /*options*
     // LAZY HACK to check for qcom hardware
-    if (strcmp(hw_module->author, "CodeAurora Forum") == 0)
-      return static_cast<mg::PlatformPriority>(mg::PlatformPriority::best + 1);
+    auto version = get_android_version();
```
Would it be good to add a try block here? so if we cant for some reason run property_get?

APokorny replied:
Like dlopen and dlsym property_get? So far we link against the library and I think the implementation does not throw.

mariogrip replied:
ok, will try tomorrow on and see if any of the devices break

mariogrip reviewed (approved):
Added small comment, other then that it LGTM :+1:

APokorny commented:
Do you think we need a prop to override the decision logic - Are there CAF + 7.1 devices without the ABI change?

mariogrip commented:
@APokorny yeah a prop would be nice. I would assume nexus/pixel devices would run into trouble since they are the only one i know thats not using caf.

APokorny committed:
Add properties to override the selection behavior

This adds code to override the default decision when
ro.build.hwc.vanilla and ro.build.hwc.qti_bsp is set

mariogrip committed:
Skip broken tests for now

fixed in version 1.x of mir, so no need to fix on this old version

mariogrip committed:
workaround for broken jenkins

UniversalSuperBox commented:
This worked on my Nexus 7, which was the only device which wasn't working before the change. Good from me!
